### PR TITLE
Try optional catch binding

### LINKED ISSUE #915: java.lang.VerifyError in language/statements/try of test262
When I enable language/statements/try in test262, I get the following error.

```
org.mozilla.javascript.tests.Test262SuiteTest > test262Case[js=language/statements/try/S12.14_A7_T1.js, opt=0, strict=false] FAILED
    java.lang.VerifyError at Test262SuiteTest.java:217

...

org.mozilla.javascript.tests.Test262SuiteTest > test262Case[js=language/statements/try/S12.14_A7_T3.js, opt=0, strict=false] FAILED
    java.lang.VerifyError at Test262SuiteTest.java:217
```
With JavaScript code like this
```js
try{
  try{
    throw "ex1";
  }finally{
    try{
      throw "ex2";
    }catch(er1){
    }
  }
}catch(er2){
}
```
I got an error like the following
```
Exception in thread "main" java.lang.VerifyError: Stack map does not match the one at exception handler 91
Exception Details:
  Location:
    org/mozilla/javascript/gen/_stdin__1._c_script_0(Lorg/mozilla/javascript/gen/_stdin__1;Lorg/mozilla/javascript/Context;Lorg/mozilla/javascript/Scriptable;Lorg/mozilla/javascript/Scriptable;[Ljava/lang/Object;)Ljava/lang/Object; @91: nop
  Reason:
    Type top (current frame, locals[5]) is not assignable to 'java/lang/Object' (stack map, locals[5])
  Current Frame:
    bci: @52
    flags: { }
    locals: { 'org/mozilla/javascript/gen/_stdin__1', 'org/mozilla/javascript/Context', 'org/mozilla/javascript/Scriptable', 'org/mozilla/javascript/Scriptable', '[Ljava/lang/Object;' }
    stack: { 'org/mozilla/javascript/JavaScriptException' }
  Stackmap Frame:
    bci: @91
    flags: { }
    locals: { 'org/mozilla/javascript/gen/_stdin__1', 'org/mozilla/javascript/Context', 'org/mozilla/javascript/Scriptable', 'org/mozilla/javascript/Scriptable', '[Ljava/lang/Object;', 'java/lang/Object', top, 'org/mozilla/javascript/Scriptable' }
    stack: { 'java/lang/Throwable' }
```
https://app.circleci.com/pipelines/github/mozilla/rhino/394/workflows/06f40ab3-873d-4ffd-b26f-753722081c8a/jobs/400/tests

I investigated the cause of this, but could not find it. If there is any way to find out, please let me know. For example, is there any way to isolate whether the StackMapTable is wrong or the bytecode itself is wrong?

### LINKED ISSUE #965: Support ES2019 Optional Catch binding
See https://github.com/tc39/proposal-optional-catch-binding

### PR DESCRIPTION
**Pull request by tuchida**
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#the_exception_identifier
```js
  try {
    JSON.parse(text);
    return true;
  } catch {
    return false;
  }
```

The omission of the exception value is implemented, but there are another error. ref. #915
We need to either fix #915 or ignore `VerifyError`.

Closes #965

### CODE REVIEW

**Comment by tuchida** [comment]
@gbrail @p-bakker Could you review it? Or should I wait for the problem to be fixed on Android?

File: src/org/mozilla/javascript/ast/CatchClause.java [hunk 1]
```diff
@@ -49,17 +48,20 @@ public Name getVarName() {
 
     /**
      * Sets catch variable node, and sets its parent to this node.
+     *
      * @param varName catch variable
      * @throws IllegalArgumentException if varName is {@code null}
```
Comment (p-bakker) [inline src/org/mozilla/javascript/ast/CatchClause.java:53]: This still holds true?

**Comment by p-bakker** [comment]
Not sure if the Android thing should hold this PR (and others) back.

Left you one little comment, most of the changes are beyond me unfortunately

File: src/org/mozilla/javascript/ast/CatchClause.java (hunk 1 above)
Comment (tuchida) [reply src/org/mozilla/javascript/ast/CatchClause.java]: 1b473d6 I fixed.

**Comment by gbrail** [comment]
This works great. I'm going to squash some commits and push manually. Thanks!

**Pull request closed without merging**